### DISCUSSION TITLE
Ssl certificate

### DISCUSSION THREAD

**Discussion by cempack** [wg-easy/wg-easy#232] [Q&A] [answered]
How Can I create a ssl certificate ? (I have a domain name)

**Comment by eandersons** [comment, marked as answer]
You could use a proxy server.

I am using nginx and my configuration is quite simple:

``` nginx
server {
    listen 80;
    listen [::]:80;
    server_name wireguard.example.org;
    return 301 https://$host$request_uri;
}

server {
    listen 443 ssl http2; 
    listen [::]:443 ssl http2;
    server_name wireguard.example.org; 

    location / { 
        proxy_pass http://127.0.0.1:51821;
        proxy_set_header Host $host; 
    }

    access_log /var/log/nginx/wireguard.example.org/access.log;
    error_log /var/log/nginx/wireguard.example.org/error.log;
}

```

Then [Certbot](https://certbot.eff.org/) can be used to easily create SSL certificates: `certbot --nginx`. Certbot will obtain and install certificates and automatically update the nginx configuration to add the necessary lines for SSL certificates to the corresponding `server` block.
